peers/open_channels.js

```diff
@@ -745,15 +831,14 @@ module.exports = (args, cbk) => {
             if (!!err) {
               return cbk(err);
             }
-
             // Consolidate all pending channels from all nodes
             const pending = flatten(res.map(n => n.pending_channels));
 
             // Only consider pending channels related to this funding tx
             const opening = pending.filter(n => n.transaction_id === txId);
 
             // Every channel to open should be reflected in a pending channel
-            if (opening.length !== toOpen.length) {
+            if ((opening.length + getTrustedChannels.length) !== toOpen.length) {
```

Comment:
This check won't work if the trusted channel takes time to show up in the channels list, to safeguard against it taking time channels should be pulled in the asyncMap above along with getPending channels. It doesn't need to look if they are trusted channels though, just that the transaction id is matching

Reply:
I added an async retry above while getting trusted channels from all channels list and that will only pass if it gets all trusted channels and will only enter this part after that.

Reply:
Ah cool ok, it still feels like overkill though

I'd just do a single call that combines getPending and getChannels where getPending is right now and then use the same logic across both result sets